src/Compiler/Checking/NicePrint.fs

```diff
@@ -925,7 +925,7 @@ module PrintTypes =
               if not denv.includeStaticParametersInTypeNames then
                   None, args
               else
-                  let regex = System.Text.RegularExpressions.Regex(@"\`\d+")
+                  let regex = System.Text.RegularExpressions.Regex(@"\`\d+", System.Text.RegularExpressions.RegexOptions.ECMAScript)
```

Comment:
See notes in OP.
By default `Regex` is Unicode aware, so char classes maps to Unicode categories and allow much more then expected here ASCII chars.

Reply:
Instead of a regular expression, this could also just be something like:

```fsharp
let tryGetTyparCount (s: string) =
    let indexOfBacktick = s.LastIndexOf '`'
    if indexOfBacktick >= 0 && indexOfBacktick < s.Length then
        match Int32.TryParse(s.AsSpan(indexOfBacktick + 1)) with
        | true, genericArgCount -> ValueSome genericArgCount
        | false, _ -> ValueNone
    else
        ValueNone
```